test/e2e/tests/privacy-mode/privacy-mode.spec.js

```diff
@@ -0,0 +1,59 @@
+const { strict: assert } = require('assert');
+const {
+  withFixtures,
+  unlockWallet,
+  defaultGanacheOptions,
+} = require('../../helpers');
+const FixtureBuilder = require('../../fixture-builder');
+
+describe('Privacy Mode', function () {
+  it('should activate privacy mode, then deactivate it', async function () {
+    await withFixtures(
+      {
+        dapp: true,
+        fixtures: new FixtureBuilder().withPreferencesController().build(),
+        ganacheOptions: defaultGanacheOptions,
+        title: this.test.fullTitle(),
+      },
+      async ({ driver }) => {
+        async function checkForHeaderValue(value) {
+          const balanceElement = await driver.findElement(
+            '[data-testid="eth-overview__primary-currency"] .currency-display-component__text',
+          );
+          const surveyText = await balanceElement.getText();
+          assert.equal(surveyText, value, `Balance should be "${value}"`);
+        }
+
+        async function checkForTokenValue(value) {
+          const balanceElement = await driver.findElement(
+            '[data-testid="multichain-token-list-item-secondary-value"]',
+          );
+          const surveyText = await balanceElement.getText();
+          assert.equal(surveyText, value, `Balance should be "${value}"`);
+        }
+
+        async function checkForPrivacy() {
+          await checkForHeaderValue('*****');
+          await checkForTokenValue('*****');
+        }
+
+        async function checkForNoPrivacy() {
+          await checkForHeaderValue('25');
+          await checkForTokenValue('25 ETH');
+        }
+
+        async function togglePrivacy() {
+          await driver.clickElement('[data-testid="sensitive-toggle"]');
+          await new Promise((resolve) => setTimeout(resolve, 2e3));
```

Comment:
This is equivalent to a `driver.delay`, which we should avoid; we should instead the element text to either be the number value or ******'s

Reply:
I've updated `togglePrivacy` to wait for text changes or a 2-second max, whichever comes first. The text change always occurs faster by milli seconds but the 2 second max is a good fallback. Lmk if you agree with the changes